content/sips/sip-307.md

```diff
@@ -67,15 +69,19 @@ Next, we describe other characteristics of the architecture.
 
 The main system proxy is an [UUPS proxy](https://eips.ethereum.org/EIPS/eip-1822) which is more gas efficient than transparent proxies by implementing the upgradeability functions in the implementation instead of the proxy. In this architecture, we develop a way of ensuring that all upgrades are further upgradeable via a special mechanism in the UpgradeModule.
 
-#### Satellites
+Since the upgradeability code in UUPS proxies is in the implemetation instead of the proxy, every update needs to make sure that the new system is upgradeable (unless of course the protocol wants to be intentionally frozen). Our implementation of UUPS proxies includes an on-chain mechanism to ensure that all new implementations are not sterile.
+
+### Associated Systems
 
-Since the deployer is 100% generic, any contract that needs to be deployed and is not a module is considered a "satellite". An example of such a contract will be the v3 SNX token. This will be a "dumb" ERC20 token, fully compliant to the ERC20 standard, that is managed by one of the system's modules and is not the system itself. All satellites will also be upgradeable through the system modules that manage them, and will also use the UUPS architecture.
+Even though the entire system will be behind a single proxy, the system may connect to other systems via its AssociatedSystems module. These associated systems may be another contract of any kind, but also other router proxy systems. Associated systems behind a router proxy may be core components of the system when used as libraries. An example associated system could be the SNX token, intended to be a simple ERC20 token with no extra code or functionality, fully compliant to the ERC20 standard. Another associated system could be a synths manager that would have the ability to update all synths in production in a single transaction.
```

Comment:
You mentioned 'libraries' before. My understanding is that you don't mean this as Solidity libraries but more abstractly?

Reply:
Yes! Will clarify.